express/components/common.py

```diff
@@ -321,6 +331,34 @@ def _create_output_dataset(
         Path(save_path).write_text(manifest.to_json(), encoding="utf-8")
         return manifest
 
+    def _verify_data_sources_in(cls, data: ExpressDataset):
+        for expected_data_source in cls.data_sources_in:
+            # get the data source from the manifest
+            if expected_data_source[0] not in data.manifest.data_sources:
+                raise ValueError(
+                    f"Input dataset is missing data source {expected_data_source}"
+                )
+            data.manifest.data_sources[expected_data_source[0]]
+            # TODO verify that the required fields are present
+            # according to the taxonomy expected_data_source[1]
+            continue
```

Comment:
This is the main part to be filled in, based on the taxonomy we define for each modality